Add Vector Database to LitAgent

jbrry opened the pull request:
### Summary :memo:
For greater forensics, and to provide contexts to FactReasoner, a vector database would be useful for storing the retrieved contents.

This PR adds a vector database tool to the `LitAgent`, and indexes those results so they can be queried later.

### Details
_Describe more what you did on changes._
- Add VectorDB tool with associated schemas for input/output for indexing and querying
- Added section in toml config file, and added a config for the tool
- Added the tool to LitAgent
- Updated LitAgent `arun` method to index documents as they are returned from the search tool.


### Checks
- Indirectly related to #14

jbrry committed:
WIP add a VectorDB to LitAgent

jbrry committed:
Add vector db tool

jbrry committed:
Linting

jbrry committed:
Merge branch 'feature/atomic-lit-agent' into feature/fact_reasoner_contexts

jbrry committed:
Remove breakpoint

NISH1001 commented on akd/agents/litsearch.py:
```diff
@@ -50,9 +62,10 @@ def __init__(
         query_agent: QueryAgent,
         extraction_agent: EstimationExtractionAgent,
         search_tool: SearxNGSearchTool,
+        vector_search_tool: VectorDBSearchTool,
```
Let's make the vetordb search optional for time being. Use only if provided.

NISH1001 commented on akd/agents/litsearch.py:
```diff
@@ -121,6 +135,39 @@ async def arun(self, params: LitAgentInputSchema) -> LitAgentOutputSchema:
+            VectorDBIndexInputSchema(
+                documents=langchain_docs,
+            ),
+        )
```
Add a check if the vectordb tool is provided, only index..else run normally without indexing

jbrry committed:
WIP prepare code for FR integration

jbrry committed:
Remove old git subtree pointing to main_lightweight

jbrry committed:
Merge commit '81efb36b93b7bd9a84f4e1cdfeb05cb1f95ea665' as 'akd/tools/fact_reasoner/fm_factual'

jbrry committed:
Squashed 'akd/tools/fact_reasoner/fm_factual/' content from commit ae7abad

git-subtree-dir: akd/tools/fact_reasoner/fm_factual
git-subtree-split: ae7abadfff1abe5aec48c607e13bac3153a71969

jbrry committed:
Remove incorrect subtree path (extra fm_factual)

jbrry committed:
Squashed 'akd/tools/fact_reasoner/' content from commit ae7abad

git-subtree-dir: akd/tools/fact_reasoner
git-subtree-split: ae7abadfff1abe5aec48c607e13bac3153a71969

jbrry committed:
Merge commit '43ac1a36cc7ef496cf6c94a0f5fd8f6af0e4ad53' as 'akd/tools/fact_reasoner'

jbrry committed:
Update paths to local subtree

jbrry committed:
Update relative paths inside subtree

jbrry committed:
Take searxng files from deepeval branch

jbrry committed:
WIP updates to FR context retrieval

jbrry committed:
Merge branch 'feature/atomic-lit-agent' into feature/fact_reasoner_contexts

jbrry committed:
Add factreasoner configs

NISH1001 commented on akd/agents/litsearch.py:
```diff
@@ -131,6 +186,20 @@ async def arun(self, params: LitAgentInputSchema) -> LitAgentOutputSchema:
+            )
+
+            print(fr_output)
+            breakpoint()
```
Remove breakpoint. For print, using `logger.debug` which is imported using `from loguru import logger`

jbrry committed:
Merge FR and Storm configs

NISH1001 committed:
Bugfix litagent

NISH1001 committed:
Refactor storm to optionally enable checkpoint

- Also change to `query` from `topic` within storm input schema
- Remove `AsyncRunMixin.ainvoke` method

movinam committed:
Add hitl for editors and checkpointing

movinam committed:
Revert memory saver

movinam committed:
Add timeout for search and return interview results

jbrry committed:
merge from develop branch
